### PR TITLE
Feature/UAR 585 continue button review update statement page

### PR DESCRIPTION
**Pull request by bmcgreevyCH**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-585

### Change description
Functionality for Radio & Continue buttons on review update statement page

### CODE REVIEW

File: src/utils/check.your.answers.ts [hunk 1]
```diff
@@ -51,6 +52,11 @@ export const postDataForReview = async (req: Request, res: Response, next: NextF
 
     const session = req.session as Session;
     const appData: ApplicationData = getApplicationData(session);
+    const noChangeReviewStatement = req.body["no_change_review_statement"];
+
+    if (noChangeReviewStatement === "0") {
```
Comment (AndreKramer7654) [inline src/utils/check.your.answers.ts:57]: For check your answers page it this the case: noChangeReviewStatement === undefined

File: test/controllers/update/update.review.statement.controller.spec.ts [hunk 2]
```diff
@@ -145,25 +156,37 @@ describe("Update review overseas entity information controller tests", () => {
...
-      // assertions dependent on UAR-587
+      const resp = await request(app)
+        .post(UPDATE_REVIEW_STATEMENT_URL)
+        .send({ no_change_review_statement: '0' });
 
-      // expect(resp.status).toEqual(302);
-      // expect(resp.header.location).toEqual(PAYMENT_LINK_JOURNEY);
+      expect(resp.status).toEqual(302);
+      expect(resp.header.location).toEqual(UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL);
     });
```
Comment (AndreKramer7654) [inline test/controllers/update/update.review.statement.controller.spec.ts:190]: Maybe test no_change_review_statement === undefined or '1'

File: src/utils/check.your.answers.ts (hunk 1 above)
Comment (bmcgreevyCH) [reply src/utils/check.your.answers.ts:57]: `  items: [
            {
              value: "1",
              text: "Yes, the information is correct",
              checked: no_change_review_statement == 1,
              attributes: {
                "data-event-id": "no-change-review-statements-radio-option-yes"
              }
            },
            {
              value: "0",
              text: "No, I need to make changes",
              checked: no_change_review_statement == 0,
              attributes: {
                "data-event-id": "no-change-review-statements-radio-option-no"
              }
            }`

Comment (AndreKramer7654) [reply src/utils/check.your.answers.ts:57]: But this is get util method is included in registration journey? So there it's undefined?

Comment (AndreKramer7654) [reply src/utils/check.your.answers.ts:57]: which should be falsy so would be ok but needs testing?

File: src/config/index.ts [hunk 3]
```diff
@@ -309,6 +309,11 @@ export const TRUSTEE_END_DATE = "#endDate-endDateDay";
 export const CORPORATE_NAME = "#corporate_name";
 export const INDIVIDUAL_NAME = "#firstName";
 export const AGENT_CODE = "#agent_code";
+export const SECURE_FILTER = "#is_secure_register";
```
Comment (n-r-b) [inline src/config/index.ts:312]: Should these be in this PR?

File: test/controllers/update/update.review.statement.controller.spec.ts [hunk 4]
```diff
@@ -141,29 +152,53 @@ describe("Update review overseas entity information controller tests", () => {
...
       mockGetApplicationData.mockReturnValueOnce(mockData);
       mockPaymentsSession.mockReturnValueOnce(PAYMENT_LINK_JOURNEY);
-      await request(app).post(UPDATE_REVIEW_STATEMENT_URL);
+      const resp = await request(app).post(UPDATE_REVIEW_STATEMENT_URL);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.header.location).toEqual(PAYMENT_LINK_JOURNEY);
+    });
+
+    test(`should redirect to UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL when noChangeReviewStatement is '0'`, async () => {
```
Comment (n-r-b) [inline test/controllers/update/update.review.statement.controller.spec.ts:176]:
```suggestion
    test(`should redirect to ${UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_PAGE} when noChangeReviewStatement is '0'`, async () => {
```

File: test/controllers/update/update.review.statement.controller.spec.ts [hunk 5]
```diff
@@ -141,29 +152,53 @@ describe("Update review overseas entity information controller tests", () => {
...
+      const mockData = {
+        ...APPLICATION_DATA_UPDATE_BO_MOCK,
+        [Transactionkey]: "",
+        [OverseasEntityKey]: "",
+      };
+      mockGetApplicationData.mockReturnValueOnce(mockData);
```
Comment (n-r-b) [inline test/controllers/update/update.review.statement.controller.spec.ts:182]: Can be replaced with:

`mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_UPDATE_BO_MOCK);`

File: src/utils/check.your.answers.ts (hunk 1 above)
Comment (n-r-b) [reply src/utils/check.your.answers.ts:57]: We also need to cover the scenario where neither of the options have been selected and we return a validation error, not sure if we're doing this?

Comment (AndreKramer7654) [reply src/utils/check.your.answers.ts:57]: That validation should be covered by the "no change statement" page so I'd expect 0/1 for update but undefined for registration.

**Review by n-r-b — APPROVED** [review]

**Pull request merged**